src/SingleCellAnnotator.imjoy.html

```diff
@@ -406,56 +418,90 @@
     },
 
     async uploadAnnotation(){
-      // Default options are marked with *
-      const sendData = JSON.parse(JSON.stringify(sendDataTemplate));
-      sendData.discarded = this.current_image_data.discarded;
-      if (this.current_image_data.discarded){
-        // delete sendData.mask;
-        sendData.mask = null;
+      // skip upload annotation if image is annotated and no new change
+      if ((this.current_image_data.annotation_status==true)&&(this.current_image_data.mask_adjusted==null)&&(this.current_image_data.location_adjusted==null)){
+        // skip uploadannotation
+        return
       }else{
-        sendData.bbox = this.current_image_data.mask_geojson.bbox;
-        sendData.location = {};
-        if (this.current_image_data.label_display_mode == 'single'){
-          sendData.location[this.current_image_data.location] = this.current_location_selected;
+        // Default options are marked with *
+        const sendData = JSON.parse(JSON.stringify(sendDataTemplate));
+        sendData.discarded = this.current_image_data.discarded;
+        if (this.current_image_data.discarded){
+          //delete sendData.mask;
+          sendData.mask = null;
+        }else{
+          sendData.bbox = this.current_image_data.mask_geojson.bbox;
+          sendData.location = {};
+          if (this.current_image_data.label_display_mode == 'single'){
+            // do not upload annotion if no location selected
+            if (this.current_location_selected==null){
+              if (this.current_image_data.mask_adjusted==null){
+                return
+              }else{
+              const anwser = await api.confirm("Please selecte Yes/No to annotate?")
+              this.current_image_data.annotation_status = null;
+              return
+              };
+            }else{
+              // upload the re-selection
+              this.current_image_data.annotation_status = false;
+            }
+            sendData.location[this.current_image_data.location] = this.current_location_selected;
           } else if (this.current_image_data.label_display_mode == 'multiple'){
             sendData.location = this.current_image_data.location;
             for (loc in sendData.location){
-              if(sendData.location[loc]==null){
-                sendData.location[loc] = false}
-            }
+              if((sendData.location[loc]==null)||(sendData.location[loc]==false)){
+                sendData.location[loc] = false
+              }else{
+                this.current_image_data.annotation_status = false}}
```

Comment:
see comments here: https://github.com/CellProfiling/cellpro-imjoy-plugins/pull/34#discussion_r409388707

Reply:
note I used null and false values. so I use === or !== in the script now

Reply:
This can be replaced with `if(!sendData.location[loc])`.

Reply:
done